test/lib/next-test-utils.ts

```diff
@@ -437,11 +438,72 @@ export function launchApp(
   )
 }
 
+/**
+ * Read a JavaScript file and return the exports.
+ *
+ * @param path The path to the JavaScript file.
+ * @returns The exports of the JavaScript file.
+ */
+function readJS(path: string) {
+  const data = readFileSync(path, 'utf8')
+  const script = new vm.Script(data, { filename: path })
+  const context = { module: { exports: {} } }
+  script.runInNewContext(context)
+  return context.module.exports
+}
+
+/**
+ * Check if the directory provides an ESLint configuration.
+ *
+ * @param dir The directory to check.
+ * @returns Whether the directory provides an ESLint configuration.
+ */
+export function shouldDisableLinting(dir: string) {
+  try {
+    // If the project already has a `next.config.js` file that includes an eslint
+    // config, then don't disable linting.
+    if (
+      existsSync(path.join(dir, 'next.config.js')) &&
+      'eslint' in readJS(path.join(dir, 'next.config.js'))
```

Comment:
Seems way to brittle to start up a VM and assume it can execute a config. I'd opt for simplicity and explicitness e.g. disable linting by default in tests and let tests opt-in if they're actually testing linting.